### PR TITLE
Add trailing note to select field options [MC-2674]

### PR DESCRIPTION
**Pull request by emrberk**

### CODE REVIEW

**Review by Sodik — CHANGES_REQUESTED** [review]

File: packages/ui/__stories__/SelectField.stories.tsx [hunk 1]
```diff
@@ -94,6 +94,11 @@ WithHelperText.args = {
   helperText: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
 }
 
+export const WithSideNote = Template.bind({})
+WithSideNote.args = {
+  options: options.map((o) => ({ ...o, sideNote: 'Side Note' })),
```
Comment (Sodik) [inline packages/ui/__stories__/SelectField.stories.tsx:99]: `{
        value: '10000',
        label: LONG_MULTIPLE_WORD_TEXT,
      },
      {
        value: '10001',
        label: LONG_ONE_WORD_TEXT,
      },`

File: packages/ui/src/Select/helpers.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { canUseDOM } from '../utils/ssr'
 export type SelectFieldOption<V> = {
   label: string
   value: V
+  sideNote?: string
```
Comment (Sodik) [inline packages/ui/src/Select/helpers.ts:8]: `trailingNote` sounds better and we have same naming in TextField - `trailingIcon`. WDYT?

File: packages/ui/src/TruncatedText.tsx [hunk 3]
```diff
@@ -42,19 +44,22 @@ export const TruncatedText: FC<TruncatedTextProps> = ({
...
+        updateToken={forceUpdateToken}
+        hoverAbleTooltip={hoverAbleTooltip}
+      >
+        {(ref) => (
+          <div ref={mergeRefs([textRef, ref])} className={cn(styles.truncatedText, className)}>
+            <span aria-labelledby={tooltip ? idTooltip : undefined}>{text}</span>
+          </div>
+        )}
+      </Tooltip>
+      {sideNote && <span className={styles.sideNote}>{sideNote}</span>}
```
[the remark this answers is not in the document]
Comment (emrberk) [reply packages/ui/src/TruncatedText.tsx]: Done 👍

File: packages/ui/__stories__/SelectField.stories.tsx (hunk 1 above)
Comment (emrberk) [reply packages/ui/__stories__/SelectField.stories.tsx]: Done 👍

File: packages/ui/src/Select/helpers.ts (hunk 2 above)
Comment (emrberk) [reply packages/ui/src/Select/helpers.ts]: I had hard times finding a name for this :) Changed it to `trailingNote` 👍

**Review by Sodik — APPROVED** [review]

**Pull request merged**